resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/BasicPersistenceTest.java

```diff
@@ -0,0 +1,187 @@
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+import uk.gov.gchq.palisade.integrationtests.resource.config.ResourceTestConfiguration;
+import uk.gov.gchq.palisade.integrationtests.resource.web.ResourceClientWrapper;
+import uk.gov.gchq.palisade.resource.LeafResource;
+import uk.gov.gchq.palisade.resource.impl.DirectoryResource;
+import uk.gov.gchq.palisade.resource.impl.FileResource;
+import uk.gov.gchq.palisade.resource.impl.SystemResource;
+import uk.gov.gchq.palisade.service.SimpleConnectionDetail;
+import uk.gov.gchq.palisade.service.resource.ResourceApplication;
+import uk.gov.gchq.palisade.service.resource.repository.JpaPersistenceLayer;
+import uk.gov.gchq.palisade.util.ResourceBuilder;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.assertThat;
+
+@EnableFeignClients
+@RunWith(SpringRunner.class)
+@Import(ResourceTestConfiguration.class)
+@SpringBootTest(classes = ResourceApplication.class, webEnvironment = WebEnvironment.DEFINED_PORT)
+@EnableJpaRepositories(basePackages = {"uk.gov.gchq.palisade.service.resource.repository"})
+@DirtiesContext(classMode = ClassMode.AFTER_CLASS)
+public class BasicPersistenceTest {
+
+    @Autowired
+    private JpaPersistenceLayer persistenceLayer;
+
+    @Autowired
+    private ResourceClientWrapper client;
+
+    /**
+     * Scenario as follows, where (F)iles, (D)irectories and (S)ystems are annotated respectively
+     *
+     *     S
+     *     |
+     *     D
+     *   / | \
+     * F   F  F
+     */
+
+    private static final SimpleConnectionDetail DETAIL = new SimpleConnectionDetail().uri("test-data-service");
+    private static final String EMPLOYEE_FORMAT = "employee";
+    private static final String CLIENT_FORMAT = "client";
+    private static final String AVRO_TYPE = "avro";
+    private static final String CSV_TYPE = "csv";
+    private static final SystemResource SYSTEM_ROOT = (SystemResource) ResourceBuilder.create("file:/");
+    private static final DirectoryResource TEST_DIRECTORY = (DirectoryResource) ResourceBuilder.create("file:/test/");
+    private static final FileResource EMPLOYEE_AVRO_FILE = ((FileResource) ResourceBuilder.create("file:/test/employee.avro"))
+            .type(AVRO_TYPE)
+            .serialisedFormat(EMPLOYEE_FORMAT)
+            .connectionDetail(DETAIL);
+    private static final FileResource EMPLOYEE_CSV_FILE = ((FileResource) ResourceBuilder.create("file:/test/employee.csv"))
+            .type(CSV_TYPE)
+            .serialisedFormat(EMPLOYEE_FORMAT)
```

Comment:
Changes will need to be made to the resources to reflect changes from the comment above.
```suggestion
            .serialisedFormat(CSV_FORMAT)
```